python/ray/workflow/tests/test_dag_to_workflow.py

```diff
@@ -108,6 +108,28 @@ def f(a):
     assert ray.get(*result3) == 10
 
 
+def test_dereference_object_refs(workflow_start_regular_shared):
+    """Ensure that object refs are dereferenced like in ray tasks."""
+
+    @ray.remote
+    def f(obj_list):
+        assert isinstance(obj_list[0], ray.ObjectRef)
+        assert ray.get(obj_list[0]) == 42
```

Comment:
`        assert ray.get(obj_list) == [42]` for better coverage which also check the list only has one element.